packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -1056,6 +1056,27 @@ export const formNestedStack = async (
   let categories = Object.keys(amplifyMeta);
 
   categories = categories.filter((category) => category !== 'providers');
+
+  const s3 = await S3.getInstance(context);
+  const currentEnvName = context.amplify.getEnvInfo().envName;
+  const urlParametersMap = {};
+  for (const category of categories) {
+    const resources = Object.keys(amplifyMeta[category]);
+    for (const resource of resources) {
+      const resourceDetails = amplifyMeta[category][resource];
+      if (resourceDetails.providerMetadata) {
+        const templateURL = resourceDetails.providerMetadata.s3TemplateURL;
+        try {
+          const key = templateURL.replace('https://s3.amazonaws.com/', '').split(resourceDetails.s3Bucket.deploymentBucketName + '/')[1];
```

Comment:
is the `replace` necessary if you're doing a split right after then taking the second item?

Reply:
You're right. I will remove it.